### PR TITLE
Add IST0137 analysis docs

### PR DESCRIPTION
**Pull request by kebe7jun**
Docs for https://github.com/istio/istio/issues/29302
[X] User Experience

### CODE REVIEW

**Comment by esnible** [comment]
@istio/wg-docs-maintainers This PR refers to an Istio limitation that is not documented in https://istio.io/latest/docs/ops/deployment/requirements/ .  Would it make sense to work with someone in @istio/wg-networking-maintainers to add a sentence/paragraph about that requirement?

**Comment by esnible** [comment]
@istio/wg-docs-maintainers I didn't get a notice this PR was created.  Would it make sense to have the _content/en/docs/reference/config/analysis/..._ folder require a sign-off from @istio/wg-user-experience-maintainers ?

File: content/en/docs/reference/config/analysis/ist0137/index.md
Comment (esnible) [inline content/en/docs/reference/config/analysis/ist0137/index.md:15]: Nit: This will be more readable if the example is as **short** as possible.  Looking at https://kubernetes.io/docs/concepts/services-networking/service/ the examples are very small, e.g.

``` yaml
apiVersion: v1
kind: Service
metadata:
  name: my-service
spec:
  ports:
    - protocol: TCP
      port: 80
      targetPort: 9376
```

Can you make your example really small?

**Review by esnible — CHANGES_REQUESTED** [review]

**Comment by frankbu** [comment]
> @istio/wg-docs-maintainers I didn't get a notice this PR was created. Would it make sense to have the content/en/docs/reference/config/analysis/... folder require a sign-off from @istio/wg-user-experience-maintainers ?

We've had an informal process, for a while now, of always requesting approval from someone in the document owner WG for any PR that is adding a new doc or making non-trivial changes. We might want to consider making this a more formal process in the future.

File: content/en/docs/reference/config/analysis/ist0137/index.md
Comment (esnible) [inline content/en/docs/reference/config/analysis/ist0137/index.md]:
```suggestion
In this example, service `nginx-a` and service `nginx-b` select the same workload `nginx` with same targetPort but different ports.
```

**Review by esnible — COMMENTED** [review]
Nearly perfect, just a grammar nit.

File: content/en/docs/reference/config/analysis/ist0137/index.md
Comment (kebe7jun) [reply content/en/docs/reference/config/analysis/ist0137/index.md]: Ok, updated.

**Review by esnible — APPROVED** [review]
Thanks!

**Review by frankbu — APPROVED** [review]

**Comment by ericvn** [comment]
@frankbu @kebe7jun 

The only problem with this PR and maybe we need to doc it better, is `data/analysis.yml` file isn't updated in this repo, but copied from https://github.com/istio/istio/blob/master/galley/pkg/config/analysis/msg/messages.yaml (at the appropriate branch (master for now)). The update reference automated job is now failing as it's trying to copy from the istio/istio file.

We should put this update in the upstream file along with a message in it to it's original location and not to update it in similar to what we have in some of the other files that are copied.

**Comment by ericvn** [comment]
It looks like the change was actually made in https://github.com/istio/istio/pull/30049, put that PR wasn't cherry-picked to the release-1.9 branch. @kebe7jun should this code be in release-1.9? If so, we should cherry-pick that PR forward which will actually update the data/analysis file. If not, then this is a istio/istio `master` branch item and should not be in istio.io yet (it's only to contain 1.9 items at this point).

**Comment by kebe7jun** [comment]
> @frankbu @kebe7jun
> 
> The only problem with this PR and maybe we need to doc it better, is `data/analysis.yml` file isn't updated in this repo, but copied from https://github.com/istio/istio/blob/master/galley/pkg/config/analysis/msg/messages.yaml (at the appropriate branch (release-1.9 for now)). The update reference automated job is now failing as it's trying to copy from the istio/istio file.
> 
> We should put this update in the upstream file along with a message in it to it's original location and not to update it in similar to what we have in some of the other files that are copied.

Sorry about that, so, what can I do to resolve this problem? revert this pr or remove the content copied from `data/analysis.yml` by a new pr?

**Comment by ericvn** [comment]
@kebe7jun I guess one needs to figure out if istio/istio#30049 should be cherry-picked to 1.9 (discussion in that PR as well). If it can be cherry-picked, then things should eventually work out. If it's not going to be cherry-picked than this PR should be reverted as the analysis for 137 isn't in release-1.9 so shouldn't be doc'd here.

**Comment by kebe7jun** [comment]
> @ericvn What do you think, @kebe7jun ? Usually I do not push hard to get analyzers to cherry-pick, as they are not functional and users can download a daily build of istioctl. If there is 0% chance of the new analyzer incorrectly offering false positive warnings we can cherry pick it.

@ericvn Oh sorry, follow @esnible , I think we should revert this pr until 1.9 is released.

**Pull request merged**
